internal/pkg/otel/components.go

```diff
@@ -95,9 +91,10 @@ func components(extensionFactories ...extension.Factory) func() (otelcol.Factori
 			fbreceiver.NewFactory(),
 			mbreceiver.NewFactory(),
 			jmxreceiver.NewFactory(),
-			kafkareceiver.NewFactory(),
 			nopreceiver.NewFactory(),
-		)
+		}
+		receivers = addExtraReceivers(receivers)
```

Comment:
Can we use a more descriptive name for this, like `addNonFipsReceivers`? The function itself should also have a docstring explaining what it does, and ideally there should be a comment here explaining why this is necessary.